### PR TITLE
feat(collections): add like button for collections

### PR DESCRIPTION
**Pull request by byhow**
re: #4626

see sample interaction at: [link](https://matters-4mb1g8yca-matters.vercel.app/@yhh354/collections/Q29sbGVjdGlvbjoxNTQ)

### CODE REVIEW

**Review by robertu7 — CHANGES_REQUESTED** [review]
Please implement as specified in the Figma design

File: src/views/User/CollectionDetail/CollectionProfile/LikeButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  LikeCollectionMutation,
+  UnlikeCollectionMutation,
+} from '~/gql/graphql'
+
+import styles from './styles.module.css'
+
+interface LikeButtonProps {
+  collection: CollectionLikeButtonPublicFragment &
+    Partial<CollectionLikeButtonPrivateFragment>
+  iconSize?: 20 | 22
```
Comment (robertu7) [inline src/views/User/CollectionDetail/CollectionProfile/LikeButton/index.tsx:28]: unused prop `iconSize`?

File: src/views/User/CollectionDetail/CollectionProfile/index.tsx
Comment (robertu7) [inline src/views/User/CollectionDetail/CollectionProfile/index.tsx:1]: L87-L96 can be removed?

image

**Review by robertu7 — APPROVED** [review]
Overall looks good

File: src/views/User/CollectionDetail/CollectionProfile/LikeButton/index.tsx (hunk 1 above)
Comment (byhow) [reply src/views/User/CollectionDetail/CollectionProfile/LikeButton/index.tsx:28]: This was not used in this component but it is used at L141

**Pull request merged**